### PR TITLE
Remove insurance products from active carts when insurance is disabled

### PR DESCRIPTION
**Pull request by Francois-Gomis**
### Reason for change

Disable Alma insurance product to remove it from active cart and delete insurance data in quote_item DB
Enable Alma insurance product when insurance is activated

[Linear task](https://linear.app/almapay/issue/ECOM-2038/ac-remove-insurance-in-cart-when-deactivated)

### Code changes

Trigger enable and disable/delete data at config save

### How to test

Create a quote with an insurance. 
Disable insurance and check product in BO
Check Cart and minicart.
Enable insurance and check product in BO

_As a reviewer, you are encouraged to test the PR locally._

### Checklist for authors and reviewers

- [ ] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [ ] The PR implements the changes asked in the referenced task / issue
- [ ] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [ ] The tests are relevant, and cover the corner/error cases, not only the happy path
- [ ] You understand the impact of this PR on existing code/features
- [ ] The changes include adequate logging and Datadog traces
- [ ] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

File: Helpers/InsuranceProductHelper.php [hunk 1]
```diff
@@ -75,11 +162,11 @@ public function createInsuranceProduct(): void
...
             "use_config_notify_stock_qty" => 0
         ]);
-        // Not visible individualy ID 1
+        // Not visible individually ID 1
         $insuranceProduct->setVisibility(Visibility::VISIBILITY_NOT_VISIBLE);
-        // ID de la classe de taxe (0 pour non applicable)
+        // Tax class ID 0 for not applicable
         $insuranceProduct->setTaxClassId(0);
-        $insuranceProduct->setDescription('Alma Insurance product');
+        $insuranceProduct->setDescription('Alma Insura      nce product');
```
Comment (Benjamin-Freoua-Alma) [inline Helpers/InsuranceProductHelper.php:169]: I think you added a space by error

File: Model/Adminhtml/Config/Insurance/ConfigurationSave.php [hunk 2]
```diff
@@ -22,46 +23,56 @@ class ConfigurationSave extends Value
...
             $arrayValue['isInsuranceActivated'] === false
         ) {
+            $this->insuranceProductHelper->disableInsuranceProductIfExist();
+            $this->quoteHelper->deleteInsuranceDataFromQuoteItemForNotConvertedQuote();
             return $this;
         }
 
         try {
-            $this->productRepository->get(InsuranceHelper::ALMA_INSURANCE_SKU);
+            $this->insuranceProductHelper->enableInsuranceProductIfExist();
```
Comment (Benjamin-Freoua-Alma) [inline Model/Adminhtml/Config/Insurance/ConfigurationSave.php:73]: You always enable the product if the insurance is activated ?
Why don't add a condition like the disable case ?
If the merchant disable manually the insurance product by chose ?

**Review by Benjamin-Freoua-Alma — COMMENTED** [review]
Not tested in QA

File: Helpers/InsuranceProductHelper.php (hunk 1 above)
Comment (Francois-Gomis) [reply Helpers/InsuranceProductHelper.php:169]: Thanks

File: Model/Adminhtml/Config/Insurance/ConfigurationSave.php (hunk 2 above)
Comment (Francois-Gomis) [reply Model/Adminhtml/Config/Insurance/ConfigurationSave.php:73]: Test is already enabled directly in service.
If merchant disable manually, impossible to add insurance in quote. Bug, need to check if product is enable for widget display

**Review by Benjamin-Freoua-Alma — APPROVED** [review]

**Review by joyet-simon — APPROVED** [review]

File: Helpers/QuoteHelper.php [hunk 3]
```diff
@@ -191,4 +201,40 @@ private function getQuoteIdFromSession():?int
...
         return $this->checkoutSession->getQuoteId();
     }
+
+
+    /**
+     * Get quote items with insurance data for quote without reserved order id and delete insurance data
+     *
+     * @return void
+     */
+    public function deleteInsuranceDataFromQuoteItemForNotConvertedQuote(): void
```
Comment (webaaz) [inline Helpers/QuoteHelper.php:211]: a bit long!

**Review by webaaz — APPROVED** [review]

**Comment by Benjamin-Freoua-Alma** [comment]
QA ok

**Pull request merged**